### PR TITLE
Role Management Listing 

### PR DESCRIPTION
**Pull request by DimalChandrasiri**

### CODE REVIEW

File: apps/admin-portal/src/components/users/roles/role-list.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                roleList && roleList.map((role, index) => (
+                    <ResourceListItem 
+                        key={ index }
+                        actions={ [
+                            /*{
+                                icon: "trash alternate",
+                                onClick: () => {},
+                                popupText: "Delete Role",
+                                type: "button"
+                            }*/
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/components/users/roles/role-list.tsx]: Shall we remove this part until the functionality is ready?

File: apps/admin-portal/src/pages/user-roles.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    const [ showWizard, setShowWizard ] = useState<boolean>(false);
+
+    useEffect(() => {
+        setListItemLimit(DEFAULT_ROLE_LIST_ITEM_LIMIT);
+    }, []);
+
+    useEffect(() => {
+        getGroupsList().then((response)=> {
+            if (response.status === 200) {
+                console.log(response.data);
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/pages/user-roles.tsx]: Let's remove this line

File: apps/admin-portal/src/pages/user-roles.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                onItemsPerPageDropdownChange={ handleItemsPerPageDropdownChange }
+                onPageChange={ handlePaginationChange }
+                /*={
+                    (
+                        <PrimaryButton onClick={ () => setShowWizard(true) }>
+                            <Icon name="add"/>
+                            Add Role
+                        </PrimaryButton>
+                    )
+                }*/
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/pages/user-roles.tsx]: Shall we remove this until we add the functionality?

File: apps/admin-portal/src/pages/user-roles.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                showPagination={ roleList?.totalResults >= DEFAULT_ROLE_LIST_ITEM_LIMIT }
+                totalPages={ Math.ceil(roleList?.totalResults / listItemLimit) }
+                totalListSize={ roleList?.totalResults }
+            >
+                <RoleList roleList={ roleList?.Resources } />
+                {
+                    showWizard && (
+                    <div>ppp</div>
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/pages/user-roles.tsx]: Can we remove this too?

File: apps/admin-portal/src/components/users/roles/role-list.tsx (hunk 1 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/src/components/users/roles/role-list.tsx]: Fixed with cccef38

File: apps/admin-portal/src/pages/user-roles.tsx (hunk 2 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/src/pages/user-roles.tsx]: Fixed with cccef38

File: apps/admin-portal/src/pages/user-roles.tsx (hunk 4 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/src/pages/user-roles.tsx]: Fixed with cccef38

File: apps/admin-portal/src/pages/user-roles.tsx (hunk 3 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/src/pages/user-roles.tsx]: Fixed with ee18200

File: apps/admin-portal/src/utils/common-utils.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ * under the License.
+ */
+
+/**
+ * This function calculate the number of days since the last
+ * modified date to the current date.
+ *
+ * @param modifiedDate
+ */
+export const handleLastModifiedDate = (modifiedDate: string): string => {
```
Comment (inthirakumaaran) [inline apps/admin-portal/src/utils/common-utils.ts]: Shall we show in months as well if the days is more than 30?

File: apps/admin-portal/src/utils/common-utils.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * This function calculate the number of days since the last
+ * modified date to the current date.
+ *
+ * @param modifiedDate
```
Comment (inthirakumaaran) [inline apps/admin-portal/src/utils/common-utils.ts]: add @param comment

File: apps/admin-portal/src/configs/routes.ts [hunk 7]
```diff
@@ -84,6 +85,16 @@ const DASHBOARD_LAYOUT_ROUTES: RouteInterface[] = [
...
+            },
+            {
+                component: UserRoles,
+                exact: true,
+                icon: "childIcon",
+                level: 2,
+                name: "User Roles",
+                path: "/roles",
+                protected: true,
+                showOnSidePanel: true,
```
Comment (jeradrutnam) [inline apps/admin-portal/src/configs/routes.ts]:
```suggestion
                showOnSidePanel: true
```

File: apps/admin-portal/src/components/users/users-list.tsx [hunk 8]
```diff
@@ -57,35 +58,10 @@ export const UsersList: React.FunctionComponent<UsersListProps> = (props: UsersL
...
-        } else {
-            return;
-        }
-    };
-
     return (
         <ResourceList className="applications-list">
             {
-                usersList && usersList.Resources && usersList.Resources instanceof Array &&
+                usersList && usersList.Resources && usersList.Resources instanceof Array &&
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/users/users-list.tsx]:
```suggestion
                usersList && usersList.Resources && usersList.Resources instanceof Array &&
```

File: apps/admin-portal/src/pages/user-roles.tsx [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+                onPageChange={ handlePaginationChange }
+                showPagination={ roleList?.totalResults >= DEFAULT_ROLE_LIST_ITEM_LIMIT }
+                totalPages={ Math.ceil(roleList?.totalResults / listItemLimit) }
+                totalListSize={ roleList?.totalResults }
+            >
+                <RoleList roleList={ roleList?.Resources } />
+            </ListLayout>
+        </PageLayout>
+    );
+}
```
Comment (jeradrutnam) [inline apps/admin-portal/src/pages/user-roles.tsx]:
```suggestion
};
```

File: apps/admin-portal/src/pages/user-roles.tsx [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+import { PrimaryButton } from "@wso2is/react-components";
+import { Icon, PaginationProps, DropdownProps } from "semantic-ui-react";
+import { DEFAULT_ROLE_LIST_ITEM_LIMIT } from "../constants";
+
+/**
+ * React component to list User Roles.
+ * 
+ * @return {JSX.Element}
+ */
+export const UserRoles = (): JSX.Element => {
```
Comment (jeradrutnam) [inline apps/admin-portal/src/pages/user-roles.tsx]: Use `ReactComponent` instead

File: apps/admin-portal/src/components/users/roles/role-list.tsx [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+import React from "react";
+import { RolesInterface } from "../../../models";
+import { ResourceList, ResourceListItem } from "@wso2is/react-components";
+import { handleLastModifiedDate } from "../../../utils";
+
+interface RoleListProps {
+    roleList: RolesInterface[]
+}
+
+export const RoleList: React.FunctionComponent<RoleListProps> = (props: RoleListProps): JSX.Element => {
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/users/roles/role-list.tsx]: Use `ReactComponent` Instead

File: apps/admin-portal/src/components/users/roles/role-list.tsx [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+                    <ResourceListItem 
+                        key={ index }
+                        itemHeader={role.displayName}
+                        metaContent={handleLastModifiedDate(role.meta.created)}
+                    />
+                ))
+            }
+        </ResourceList>
+    );
+}
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/users/roles/role-list.tsx]:
```suggestion
};
```

File: apps/admin-portal/src/models/roles.ts [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+export interface RolesInterface {
+    displayName: string;
+    id: string;
+    meta: RolesMetaInterface;
+    members: RolesMemberInterface[];
+}
+
+/**
+ * 
+ */
```
Comment (jeradrutnam) [inline apps/admin-portal/src/models/roles.ts]: Empty comment block

File: apps/admin-portal/src/utils/common-utils.ts (hunk 5 above)
Comment (jeradrutnam) [reply apps/admin-portal/src/utils/common-utils.ts]: Shall we use moment js for this, https://www.npmjs.com/package/moment

Comment (DimalChandrasiri) [reply apps/admin-portal/src/utils/common-utils.ts]: Fixed with commit 184e209. Used MomentJs.

File: apps/admin-portal/src/utils/common-utils.ts (hunk 6 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/src/utils/common-utils.ts]: fixed with commit 184e209.

File: apps/admin-portal/src/models/roles.ts (hunk 13 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/src/models/roles.ts]: fixed with commit 184e209.

File: apps/admin-portal/src/pages/user-roles.tsx (hunk 10 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/src/pages/user-roles.tsx]: Fixed with commit 1336bf3. Used `ReactElement` as the return type.

File: apps/admin-portal/src/components/users/roles/role-list.tsx (hunk 11 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/src/components/users/roles/role-list.tsx]: Fixed with commit 1336bf3. Used `ReactElement` as the return type.

**Pull request merged**